88972898 array homework 1 master

anthonyfalzetti opened the pull request.

anthonyfalzetti committed:
While not complete this is an excellent start to the homework.

I am hitting a wall with the space between objects in an array. I can complete this homework,
but it will be sloppy and I am sure there is a better and cleaner way to complete it. I shall return
to conquer it soon.

[#88972898]

anthonyfalzetti committed:
Completed home for arrays

Was told that my slopppy .join would be sufficent so moving on to new homework.

[#88972898]

jasonnoble commented on arrays/array_practice.rb:
```diff
@@ -0,0 +1,22 @@
+puts "The second from last element is #{starting_array[-2]}"
+puts "The first four elements are '#{starting_array.first(4).join(', ')}'"
+print 'If we delete 5, 6 and 7 from the array,'
+puts " we're left with [#{new_array.join(',')}]"
```
You didn't delete 5, 6, 7 from starting_array, you created a new array...

jasonnoble commented on arrays/array_practice.rb:
```diff
@@ -0,0 +1,22 @@
+puts "The element with an index of 3 is #{starting_array[3]}"
+puts "The second from last element is #{starting_array[-2]}"
+puts "The first four elements are '#{starting_array.first(4).join(', ')}'"
+print 'If we delete 5, 6 and 7 from the array,'
```
Try `delete_if`

jasonnoble commented on arrays/array_practice.rb:
```diff
@@ -0,0 +1,22 @@
+puts " we're left with [#{new_array.unshift(5).join(',')}] "
+print 'If we add 6 at the end of the array,'
+puts " we're left with [#{new_array.push(6).join(',')}]"
+puts "Only the elements #{new_array.keep_if { |number| number > 8 }} are > 8."
```
This modifies the array, can you find a method that doesn't modify the original array?

anthonyfalzetti committed:
Update the array_practice.rb

I removed the extra array and instead ran all the methods on the existing one.
I changed the .reject to delete_if and I changed the keep to a select so as not to change the array.
I change dthe iterators from number to digit... so that I wasn't over the space limit :)

[#88972898]
